frameworks/Rust/may-httparse/src/main.rs

```diff
@@ -0,0 +1,72 @@
+extern crate may;
+extern crate httparse;
+extern crate bytes;
+extern crate time;
+#[macro_use] extern crate serde_json;
+
+use may::coroutine;
+use may::net::TcpListener;
+use httparse::Status;
+use bytes::BufMut;
+use std::io::{Read, Write};
+
+fn req_parse(buf: &[u8], path: &mut String) -> Option<usize> {
+    let mut headers = [httparse::EMPTY_HEADER; 16];
+    let mut req = httparse::Request::new(&mut headers);
+
+    if let Ok(Status::Complete(i)) = req.parse(buf) {
+        path.clear();
+        path.push_str(req.path.unwrap_or("/"));
+        return Some(i);
+    }
+
+    None
+}
+
+fn main() {
+    may::config().set_io_workers(4);
```

Comment:
use number of cpus here, e.g. https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/Rust/hyper/src/main.rs#L65